client/src/components/molecules/EditForm/ItemCreateForm.js

```diff
@@ -71,28 +94,51 @@ export const ItemCreateForm = (data) => {
           </StyledLabel>
           <StyledError name="description" component="div" />
 
-          <CategoryFields />
+          <CategoryFields onCategoryChange={handleCategoryChange} />
 
           <StyledLabel>
             Brand
             <StyledInput name="brand" />
           </StyledLabel>
           <StyledError name="brand" component="div" />
 
-          <StyledLabel>
-            Clothing size
-            <StyledCheckboxGroup
-              name="clothingSize"
-              options={clothingSizeOptions}
-            />
-          </StyledLabel>
-          <StyledError name="clothingSize" component="div" />
+          {showBatchOptions ? (
+            <>
+              {selectedCategory === 'shoes' && <ShoeSizeFields />}
+              {selectedCategory === 'children' &&
+                selectedSubCategory === 'baby' && (
+                  <>
+                    <ClothingSizeFields />
+                    <ShoeSizeFields />
+                  </>
+                )}
+              {selectedCategory !== 'shoes' &&
+                !(
+                  selectedCategory === 'children' &&
+                  selectedSubCategory === 'baby'
+                ) && <ClothingSizeFields />}
+            </>
```

Comment:
you might want to extract this to a function above called something like `renderBatchOptions` - would give you a bit more freedom with the conditional logic